src/with-next-video.ts

```diff
@@ -33,11 +33,21 @@ export function withNextVideo(nextConfig: any, videoConfig?: VideoConfig) {
     const VIDEOS_PATH = join(process.cwd(), folder);
     const TMP_PUBLIC_VIDEOS_PATH = join(process.cwd(), 'public', `_next-video`);
 
-    symlinkDir(VIDEOS_PATH, TMP_PUBLIC_VIDEOS_PATH);
-
-    process.on('exit', async () => {
-      await fs.unlink(TMP_PUBLIC_VIDEOS_PATH);
-    });
+    try {
+      const files = fs.readdirSync(VIDEOS_PATH);
+      if (files.length > 0) {
+        symlinkDir(VIDEOS_PATH, TMP_PUBLIC_VIDEOS_PATH);
+        process.on('exit', () => {
+          try {
+            fs.unlinkSync(TMP_PUBLIC_VIDEOS_PATH);
+          } catch (error) {
+            // Silently fail if unable to remove symlink
+          }
+        });
+      }
+    } catch (error) {
+      // Videos directory doesn't exist or is not accessible, which we treat the same as having no files
+    }
```

Comment:
hey Liam, thanks for contributing!

this works when there initially are no video files but if I keep the Next.js app running and then add a video file to the folder there wouldn't be a symlink when refreshing the page.

the symlink is needed temporarily before the asset is uploaded / processed to show the raw local video file in the player in the webpage. only in dev mode.

Reply:
hey Wesley, that makes sense! Yeah, sorry I missed that. But, it looks like this might not be a valid issue, so I'll just close this.

I've been looking for projects to contribute to, so if you have any suggestions, let me know!